src/main/java/com/zaxxer/nuprocess/osx/OsxProcess.java

```diff
@@ -181,14 +181,14 @@ private int spawnWithCwd(final IntByReference restrict_pid,
       Pointer oldCwd = new Pointer(peer);
       LibC.getcwd(oldCwd, cwdBufSize);
       String newCwd = cwd.toAbsolutePath().toString();
-      int rc = LibC.SYSCALL.syscall(SyscallLibrary.SYS___pthread_chdir, newCwd);
+      int rc = LibC.SYSCALL.syscall(LibC.SYS___pthread_chdir, newCwd);
       checkReturnCode(rc, "syscall(SYS__pthread_chdir) failed to set current directory");
 
       try {
          return LibC.posix_spawnp(restrict_pid, restrict_path, file_actions, restrict_attrp, argv, envp);
       }
       finally {
-         rc = LibC.SYSCALL.syscall(SyscallLibrary.SYS___pthread_chdir, oldCwd);
+         rc = LibC.SYSCALL.syscall(LibC.SYS___pthread_chdir, oldCwd);
```

Comment:
Have you tested the above two changes on MacOS X to ensure compatibility?

Reply:
The test case RunTest.changeCwd() passes on macOS 10.15.4, it appears that the syscall.h for OSX has the value of 348 specified (https://github.com/phracker/MacOSX-SDKs/blob/master/MacOSX10.15.sdk/usr/include/sys/syscall.h) whereas the FreeBSD syscall.h does not.  Also worth noting that the FreeBSD value in syscall.h is SYS_chdir, there is not a SYS___pthread_chdir in the header like there is in the OSX version.  I am not sure of the impact of using the traditional SYS_chdir vs SYS___pthread_chdir on FreeBSD, but I can confirm the change does work on FreeBSD 11.3.

Reply:
Given clearer understanding of the change, I actually expect issues under load.  However, there is no multi-threaded load-test version of the changeCwd() test.  One needs to be written.

Here is how the current (pthread) code works on OS X:
```
Thread 1: pthread_chdir /pathA
Thread 1: spawn process ... process inherits cwd of /pathA
```
Now multi-threaded interleaved:
```
Thread 1: pthread_chdir /pathA
Thread 2: pthread_chdir /pathB
Thread 1: spawn process ... process inherits cwd of /pathA
Thread 2: spawn process ... process inherits cwd of /pathB
```
Because SYS_chdir is global rather than thread specific, here what I expect to happen with multiple interleaved threads:
```
Thread 1: SYS_chdir /pathA
Thread 2: SYS_chdir /pathB
Thread 1: spawn process ... process inherits cwd of /pathB
Thread 2: spawn process ... process inherits cwd of /pathB
```
The process spawned by thread 1 will have the incorrect cwd.

I would need to see a test that convinces me that is not the case before I can accept the change.

Reply:
Brett -

I'll work on a test case to implement the above, I did quite a bit of digging through the FreeBSD source to find a pthread equivalent chdir call, and couldn't find anything, more research to be done, but I wonder if somehow freebsd is handling this differently.  We've replaced the library in bitbucket with the version that implements this change, and it is a quite active server and we have thus far not experienced any issues. 

But as I said, I'll implement the test case and run the unit tests on FreeBSD and see what the results are. 

Jason

Reply:
Alternatively, you can add a conditional path in the code, one for FreeBSD and one for OSX, and roll the dice on FreeBSD.

Reply:
That is more or less what I am doing; i'm only changing the value of SYS___pthread_chdir to 12 in LibC if the os is Freebsd, otherwise it remains 348 (including on Mac), so in a sense, I'm rolling the dice now 😊 

I feel like there is more to this puzzle with freebsd and pthread, and just haven't had the time to dig further into it... I agree with your test case and think that is a good spot to start. If the multithreaded load test proves successful on FreeBSD it would indicate its doing something under the hood with pthread and somehow mapping the SYS_chdir to the executing thread as oppose to the process... If it fails, then I'm missing something and need to dig deeper. Unfortunately the standard simple google queries provide no details on this, so its a dig through the FreeBSD source kind of issue, my favorite haha